src/Processors/Formats/Impl/ParquetFileMetaDataCache.cpp

```diff
@@ -0,0 +1,20 @@
+#include <Processors/Formats/Impl/ParquetFileMetaDataCache.h>
+
+#ifdef USE_PARQUET
+
+namespace DB
+{
+
+ParquetFileMetaDataCache::ParquetFileMetaDataCache()
```

Comment:
This looks too silly

Reply:
Why do we need this singleton here? Can't the cache be a local member somewhere?

Reply:
It is used by two classes: `ParquetBlockInputFormat` and `ParquetMetadataInputFormat`